Chisel/CHLDuplicateDetection.m

```diff
@@ -0,0 +1,49 @@
+//
+//  CHLDuplicateDetection.m
+//  Chisel
+//
+//  Copyright © 2016 Facebook. All rights reserved.
+//
+
+#if __has_feature(objc_arc)
+#error CHLObjectDuplicateDetection.m expects ARC to be disabled.
+#endif
+
+#import "CHLDuplicateDetection.h"
+
+#import "CHLObjectAllocations.h"
+
+#import <objc/runtime.h>
+
+NSArray *CHLFindDuplicates(Class cls, BOOL (^equalFunction)(id left, id right)) {
+  NSMutableArray *objects = [NSMutableArray new];
+  
+  CHLEnumerateObjectsWithBlock(^(id object) {
+    if (object_getClass(object) == cls) {
```

Comment:
I wonder if we should use `[object isKindOfClass:cls]` here? It will be pretty common to look for duplicates of `NSString`, `NSArray`, `NSDictionary`, etc, but those are all implemented as class clusters and as such are essentially abstract classes and no instances will be found. These collection objects will actually be internal subclasses, and I we definitely don't want to force callers to refer to internal subclasses for many reasons.

On the flip side, there might be cases where the caller wants to find duplicates based on exact class matching, not subclasses. I suspect this will be the less common case, so I would be fine with not supporting this case right now, and dealing with it later.